### PR TITLE
#685 Re-implement specific fields for student information

### PR DESCRIPTION
**Pull request by kevinhowbrook**
Ticket [685](https://projects.torchbox.com/projects/rca-website-rebuild/tickets/685)

This PR reworks some fields on the StudentPage model so we can render them on a form for users to update their own page. The rationale is that streamfield on a Django view/form won't behave well so it's better to be more specific where we can

### CODE REVIEW

File: rca/people/models.py [hunk 1]
```diff
@@ -504,6 +504,21 @@ class RelatedStudentPage(Orderable):
     panels = [PageChooserPanel("page")]
 
 
+class StudentPageGallerySlide(Orderable):
+    source_page = ParentalKey("StudentPage", related_name="gallery_slides")
+    title = models.CharField(max_length=120)
+    image = models.ForeignKey(
```
Comment (richbrennan) [inline rca/people/models.py:510]: Should image be `blank=True`?

File: rca/project_styleguide/templates/patterns/molecules/accordion/accordion.html [hunk 2]
```diff
@@ -40,7 +40,7 @@ <h5 class="accordion__heading heading heading--five">{{ accordion.heading }}</h5
                     {{ accordion.introduction }}
                 </div>
             {% endif %}
-            {{ accordion.body|richtext }}
+            {{ accordion.body|richtext|linebreaks }}
```
Comment (richbrennan) [inline rca/project_styleguide/templates/patterns/molecules/accordion/accordion.html:43]: Is `linebreaks` needed for existing data...?

File: rca/people/models.py [hunk 3]
```diff
@@ -660,13 +700,83 @@ def name(self):
...
+                    }
+                }
+            )
+        return data
+
+    @property
+    def student_gallery(self):
+        # Format related model to a nice dict
+        data = []
+        if not self.gallery_slides.exists():
```
[the remark this answers is not in the document]
Comment (kevinhowbrook) [reply rca/people/models.py]: Good point, I've removed it, the `for item in self.gallery_slides.all():` will handle it. I have an unhealthy fixation on trying to return early :rofl:

File: rca/people/models.py [hunk 4]
```diff
@@ -660,13 +700,83 @@ def name(self):
...
+                        "author": item.author,
+                        "image": item.image,
+                    }
+                }
+            )
+        return data
+
+    @property
+    def student_related_links(self):
+        if not self.relatedlinks.exists():
```
[the remark this answers is not in the document]
Comment (kevinhowbrook) [reply rca/people/models.py]: removed

File: rca/project_styleguide/templates/patterns/molecules/accordion/accordion.html (hunk 2 above)
Comment (kevinhowbrook) [reply rca/project_styleguide/templates/patterns/molecules/accordion/accordion.html:43]: Linebreaks is actually added to handle these new fields, as they aren't rich text. But the accordion is used allover the place with rich text fields.

**Pull request merged**